#!/usr/bin/env python

PACKAGE = "range_dbscan_segment"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()


gen.add("detect_min",             double_t, 0, "Default: 1",      1,    0,  2)
gen.add("detect_max",             double_t, 0, "Default: 10",     10,    2,  20)
  
gen.add("eps_coeff",              double_t, 0, "Default: 1",     0.5,    1,  3)
gen.add("CorePointMinPt",         int_t, 0,    "Default: 25",      25,   10,   100)
gen.add("ClusterTolerance",       double_t, 0, "Default: 0.3",     0.3,   0,   1)
gen.add("MinClusterSize",         int_t, 0,    "Default: 30",      30,   10,   100)
gen.add("MaxClusterSize",         int_t, 0,    "Default: 1000",    1000,   20,   10000)

gen.add("segmentTh_H",           double_t, 0, "Default: 20",     20,    0,  70)
gen.add("segmentTh_V",           double_t, 0, "Default: 10",     10,    0,  70)

gen.add("d_th",                   double_t, 0, "Default: 0.1",    0.1,    0,  1)


exit(gen.generate(PACKAGE, "range_dbscan_segment", "range_dbscan_segment_"))